pkg/app/middleware_test.go

```diff
@@ -73,3 +77,61 @@ var _ = Describe("WithHealthMiddleware", func() {
 		Expect(resp.StatusCode).To(Equal(http.StatusOK))
 	})
 })
+
+var _ = Describe("WithInitrdViaHTTPMiddleware", func() {
```

Comment:
Should we add a test here for the TLS condition?

Reply:
Good idea, added